libavformat/dashdec.c

```diff
@@ -2368,12 +2372,29 @@ static int dash_read_packet(AVFormatContext *s, AVPacket *pkt)
     if (!cur) {
         return AVERROR_INVALIDDATA;
     }
+
     while (!ff_check_interrupt(c->interrupt_callback) && !ret) {
         ret = av_read_frame(cur->ctx, pkt);
         if (ret >= 0) {
             /* If we got a packet, return it */
             cur->cur_timestamp = av_rescale(pkt->pts, (int64_t)cur->ctx->streams[0]->time_base.num * 90000, cur->ctx->streams[0]->time_base.den);
             pkt->stream_index = cur->stream_index;
+
+            av_dict_set_int(&metadata_dict, "segNumber", cur->cur_seq_no, 0);
+            av_dict_set_int(&metadata_dict, "segSize", cur->cur_seg_size, 0);
```

Comment:
Is `cur_seg_size `the same as `videos[x]->cur_seg->size`?

Reply:
It appears that way.  On line 1837 whenever `openInput` is called, the representation cur_seg_size gets set to the fragment size.

Reply:
There are a few other spots that modify `size` which I can get from the representation object we're looking at, so I'll change it to use that instead.